fix Amazon system-release version parsing

jmahler opened the pull request:
Previously it was assumed that the Amazon system-release
number was the final value of the string.  This isn't always
the case.  Some releases have the name at the end.

  Amazon Linux release 2
  Amazon Linux release 2 (Karoo)

Fix by instead looking for a number in the string.

##### SUMMARY

Fixes #48823

##### ISSUE TYPE

- Bugfix Pull Request

samdoran commented on lib/ansible/module_utils/facts/system/distribution.py:
```diff
@@ -228,10 +228,11 @@ def parse_distribution_file_Slackware(self, name, data, path, collected_facts):
+            return False, amazon_facts
         amazon_facts['distribution'] = 'Amazon'
-        amazon_facts['distribution_version'] = data.split()[-1]
+        version = re.search(r" ([\d\.]+) ?", data)
```
Rather than use a regexp, split the string and get the digit: 
```suggestion
        version = [n for n in data.split() if n.isdigit()]
```

samdoran commented on lib/ansible/module_utils/facts/system/distribution.py:
```diff
@@ -228,10 +228,11 @@ def parse_distribution_file_Slackware(self, name, data, path, collected_facts):
-        amazon_facts['distribution_version'] = data.split()[-1]
+        version = re.search(r" ([\d\.]+) ?", data)
+        if version:
+            amazon_facts['distribution_version'] = version.group(1)
```
```suggestion
            amazon_facts['distribution_version'] = version[0]
```

samdoran commented:
Don't worry about the test failures unrelated to your change. They are frustrating but won't prevent this form being merged.

Please create a changelog fragment. See [this fragment](https://github.com/ansible/ansible/blob/stable-2.7/changelogs/fragments/reboot-change-default-boot-command.yaml) as an example.

samdoran commented on lib/ansible/module_utils/facts/system/distribution.py:
```diff
@@ -228,10 +228,11 @@ def parse_distribution_file_Slackware(self, name, data, path, collected_facts):
         amazon_facts['distribution'] = 'Amazon'
-        amazon_facts['distribution_version'] = data.split()[-1]
+        version = re.search(r" ([\d\.]+) ?", data)
+        if version:
```
We always want to return something for distribution version:
```suggestion
        amazon_facts['distribution_version'] = 'NA'
        if version:
```

jmahler committed:
fix Amazon system-release version parsing

Previously it was assumed that the Amazon system-release
number was the final value of the string.  This isn't always
the case.  Some releases have the name at the end.

  Amazon Linux release 2
  Amazon Linux release 2 (Karoo)

Fix by instead looking for a number in the string.

Fixes #48823

jmahler commented:
Re: samdoran,

Good suggestions, updated in v2.

The only slight difference from your changes I made was to
use a ternary operation to ensure a valid version.

samdoran commented:
rebuild_merge
